utils.py

```diff
@@ -809,6 +809,39 @@ def polylcm(a, b):
         lcm_roots = a_roots + b_roots
         return numpy.poly1d(lcm_roots, True)
 
+def multi_polylcm(P):
+    roots_list = [i.r.tolist() for i in P]
+    roots_by_mult = []
+    lcm_roots_by_mult = []
+    for roots in roots_list:
+        root_builder = []
+        for root in roots:
+            repeated = False
+            for i in range(len(root_builder)):
+                if abs(root_builder[i][0]-root) < 10**-3:
+                    root_builder[i][1] += 1
+                    repeated = True
+                    break
+            if not repeated:
```

Comment:
This pattern of using a flag is why Python's `for` has an `else`. http://book.pythontips.com/en/latest/for_-_else.html#else-clause